python/cudf/cudf/tests/test_orc.py

```diff
@@ -192,6 +192,48 @@ def test_orc_reader_strings(datadir):
     assert_eq(expect, got, check_categorical=False)
 
 
+def test_orc_read_statistics(datadir):
+    path = datadir / "TestOrcFile.testStripeLevelStats.orc"
```

Comment:
I see that `try/except` is used in other parts of this test but this seems strange.  I would recommend removing that an ensuring the test always.  Is there some other context I'm missing ?  cc @shwina 

Generally, it would also be good to generate this file rather than check in data.  If we construct the data it helps when reading the assertions below

Reply:
I'm not sure why it's being done this way. Agree that if possible we should avoid this. Is it possible to use something like `StringIO` or `BytesIO` here instead of an actual file?

Reply:
> I see that try/except is used in other parts of this test but this seems strange. I would recommend removing that an ensuring the test always. Is there some other context I'm missing ? 

I was just following the pattern of other tests in this file. Maybe the idea was that what/which files are in `data/orc` is subject to change and it is convenient for tests to get automatically skipped if they make use of data that isn't there.

> Generally, it would also be good to generate this file rather than check in data. If we construct the data it helps when reading the assertions below

ORC writer support in Python seems poor to me compared to Java. I wouldn't know what ORC writer to use and then what if a test started failing because the ORC writer dropped support for some kind of statistics (statistics are optional and there are multiple (I think 2) versions of them in the ORC format). Instead, I added a short description of the data - let me know if that is sufficient. By the way, this file is from Apache ORC repo [1] by the way.

[1] https://github.com/apache/orc/blob/master/examples/

Reply:
Maybe that file is ok.  I was thinking something along the lines of:

```python
In [3]: cdf = cudf.datasets.randomdata()

In [4]: cdf.to_orc('/tmp/tmpfile.orc')

In [5]: cudf.read_orc('/tmp/tmpfile.orc').head()
Out[5]:
     id         x         y
0   938 -0.722027 -0.225432
1  1060 -0.020923  0.267059
2  1010  0.157010  0.843920
3   979  0.840437 -0.256838
4  1030  0.961811  0.349156
```

Reply:
I suppose I could use cuDF's ORC writer. After all, it does support writing statistics for int and string which are the only data types the file has. Just seems better to not depend on a large component like the cuDF ORC writer. If the ORC writer has bugs (like maybe this one [1]?), then that effects this test.

[1] https://github.com/rapidsai/cudf/issues/5826

Reply:
Ah, then perhaps it's reasonable to add the file in while #5826 is fixed. Thanks @calebwin